src/scenario/area.hpp

```diff
@@ -41,6 +42,17 @@ class cArea {
 		, terrain(dim, dim)
 		, maps(dim, boost::dynamic_bitset<>(dim))
 	{}
+
+	bool is_on_map(location loc) const
+	{
+		return loc.x < max_dim && loc.y < max_dim && loc.x >= 0 && loc.y >= 0;
+	}
+
+	bool is_tile_on_edge(location loc) const
```

Comment:
Between this and the above function, I think their names should be consistent - either both include "tile" or neither does. (I tend to prefer shorter names, but I guess the difference in this case is fairly small.)

Reply:
Doesn't this code check if the tile is on the map (within the boundaries specified by the minmax)? I think this name is wrong, let me know if you agree.

Reply:
Hmm. Let me analyze the possible cases for `x`, supposing for simplicity that `max_dim = 8`...

* `x < 0`, ie it's not on the map
-- `minmax(0, 7, x)` yields 0, which is not equal to x
-- return false
* `x = 0`, ie it's on the edge
-- `minmax(0, 7, 0)` yields 0, which is equal to x
-- return true
* `x = 5`, ie it's in the middle somewhere
-- `minmax(0, 7, 5)` yields 5, which is equal to x
-- return true
* `x = 7`, ie it's on the edge
-- `minmax(0, 7, 7)` yields 7, which is equal to x
-- return true
* `x >= 8`, ie it's off the map
-- `minmax(0, 7, x)` yields 7, which is not equal to x
-- return false

...so it looks like you're correct and this formula is actually equivalent to the one above, so you can pretty much remove this and use `is_on_map` instead.

Reply:
(Assuming minmax's signature is minmax(min, max, val), but anything else wouldn't make sense in context.)